# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1411240705 14400
#      Sat Sep 20 15:18:25 2014 -0400
# Node ID 5bc7d059344db214bfc5c1d5cd350da394e32c4d
# Parent  fbe5a2dd64aea45301ee8519cce293bd584e4c87
# Parent  2036e17079870803c229daf85a5de150fb87a358
maint: Periodic merge of gui-release to default.

diff --git a/.hgsubstate b/.hgsubstate
--- a/.hgsubstate
+++ b/.hgsubstate
@@ -1,1 +1,1 @@
-6d4e36653a40da6604507406f2a97e3e64bf9dbf gnulib-hg
+8728cf80b7d84b0ac401debac5b37217aac0d2e2 gnulib-hg
diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -1018,17 +1018,17 @@ file_editor::handle_editor_state_changed
                                           bool is_octave_file)
 {
   // In case there is some scenario where traffic could be coming from
   // all the file editor tabs, just process info from the current active tab.
   if (sender () == _tab_widget->currentWidget ())
     {
       _copy_action->setEnabled (copy_available);
       _cut_action->setEnabled (copy_available);
-      _run_selection_action->setEnabled (copy_available && is_octave_file);
+      _run_selection_action->setEnabled (copy_available);
       _run_action->setEnabled (is_octave_file);
 
       if (!file_name.isEmpty ())
         {
           ced = QDir::cleanPath (file_name);
           int lastslash = ced.lastIndexOf ('/');
 
           // Test against > 0 because if somehow the directory is "/" the
diff --git a/scripts/startup/inputrc b/scripts/startup/inputrc
--- a/scripts/startup/inputrc
+++ b/scripts/startup/inputrc
@@ -19,8 +19,13 @@
 ##   sequence for the DOWN arrow.
 
 "\e[B": history-search-forward
 
 ## Disable so the usual paste shortcut will work on Windows systems.
 ## \C-q should still be available for quoted insert.
 
 "\C-v": ""
+
+## In windows, disable audiable bell which is enabled by default
+$if term=cygwin
+   set bell-style none
+$endif
